src/engine/strat_engine/range_alloc.rs

```diff
@@ -0,0 +1,232 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::cmp::min;
+use std::u64::MAX;
+
+use devicemapper::Sectors;
+use ranged_set::RangedSet;
+
+#[derive(Debug)]
+pub struct RangeAllocator {
+    limit: Sectors,
+    used: RangedSet<u64>,
+}
+
+impl RangeAllocator {
+    /// Create a new RangeAllocator with all entries in the range unused.
+    pub fn new(limit: Sectors) -> RangeAllocator {
+        assert!(*limit < MAX, "limit must be less than 2^64");
```

Comment:
yeah, limit=25 means there are 25 entries in the range, numbered 0 to 24. What's a better word to use here?